Still walking through the test plan (more feedback to come shortly...), but when I switched my locale to de_DE, deleted it and restarted the dev container, I got a stack trace of:
```
Traceback (most recent call last):
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/flask/app.py", line 2091, in __call__
    return self.wsgi_app(environ, start_response)
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/flask/app.py", line 2076, in wsgi_app
    response = self.handle_exception(e)
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/user/github/freedomofpress/securedrop/securedrop/source_app/main.py", line 38, in index
    return render_template('index.html')
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/flask/templating.py", line 147, in render_template
    return _render(
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/flask/templating.py", line 128, in _render
    rv = template.render(context)
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/jinja2/environment.py", line 1304, in render
    self.environment.handle_exception()
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/jinja2/environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/home/user/github/freedomofpress/securedrop/securedrop/source_templates/index.html", line 66, in top-level template code
    {% include 'locales.html' %}
  File "/home/user/github/freedomofpress/securedrop/securedrop/source_templates/locales.html", line 4, in top-level template code
    {{ gettext('Selected language')}}: {{ g.locales[g.localeinfo.id].display_name }}
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/jinja2/environment.py", line 474, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'collections.OrderedDict object' has no attribute 'de_DE'
```

Resetting my Tor Browser identity/session fixed it so I don't think it's a problem in practice for production installs, but flagging it as it seems to reveal another place not using/checking USABLE_LOCALES first.

--- a/securedrop/i18n.py
+++ b/securedrop/i18n.py
@@ -17,7 +17,7 @@
 #
 import collections
 
-from typing import Dict, List, Set
+from typing import List, Set
 
 from babel.core import (
     Locale,
@@ -177,22 +177,19 @@ def map_locale_display_names(config: SDConfig) -> None:
     to distinguish them. For languages with more than one translation,
     like Chinese, we do need the additional detail.
     """
-    language_locale_counts = collections.defaultdict(int)  # type: Dict[str, int]
-    for l in sorted(config.SUPPORTED_LOCALES):
-        if Locale.parse(l) not in USABLE_LOCALES:
-            continue
-
-        locale = RequestLocaleInfo(l)
-        language_locale_counts[locale.language] += 1
-
+    seen: Set[str] = set()
     locale_map = collections.OrderedDict()
     for l in sorted(config.SUPPORTED_LOCALES):
         if Locale.parse(l) not in USABLE_LOCALES:
             continue
 
         locale = RequestLocaleInfo(l)
-        if language_locale_counts[locale.language] > 1:
+        if locale.language in seen:
+            # Disambiguate translations for this language.
             locale.use_display_name = True
+        else:
+            seen.add(locale.language)
+
         locale_map[str(locale)] = locale
 
     global LOCALES

--- a/securedrop/i18n.py
+++ b/securedrop/i18n.py
@@ -179,11 +179,17 @@ def map_locale_display_names(config: SDConfig) -> None:
     """
     language_locale_counts = collections.defaultdict(int)  # type: Dict[str, int]
     for l in sorted(config.SUPPORTED_LOCALES):
+        if Locale.parse(l) not in USABLE_LOCALES:
+            continue
+
         locale = RequestLocaleInfo(l)
         language_locale_counts[locale.language] += 1
 
     locale_map = collections.OrderedDict()
     for l in sorted(config.SUPPORTED_LOCALES):
+        if Locale.parse(l) not in USABLE_LOCALES:
+            continue
+
         locale = RequestLocaleInfo(l)
         if language_locale_counts[locale.language] > 1:
             locale.use_display_name = True

--- a/securedrop/i18n.py
+++ b/securedrop/i18n.py
@@ -17,7 +17,7 @@
 #
 import collections
 
-from typing import Dict, List
+from typing import Dict, List, Set
 
 from babel.core import (
     Locale,
@@ -29,7 +29,7 @@ from babel.core import (
 from flask import Flask, g, request, session
 from flask_babel import Babel
 
-from sdconfig import SDConfig
+from sdconfig import SDConfig, FALLBACK_LOCALE
 
 
 class RequestLocaleInfo:
@@ -126,32 +126,46 @@ def configure_babel(config: SDConfig, app: Flask) -> Babel:
     return babel
 
 
+def parse_locale_set(codes: List[str]) -> Set[Locale]:
+    return {Locale.parse(code) for code in codes}
+
+
 def validate_locale_configuration(config: SDConfig, babel: Babel) -> None:
     """
-    Ensure that the configured locales are valid and translated.
+    Check that configured locales are available in the filesystem and therefore usable by
+    Babel.  Warn about configured locales that are not usable, unless we're left with
+    no usable default or fallback locale, in which case raise an exception.
     """
-    if config.DEFAULT_LOCALE not in config.SUPPORTED_LOCALES:
-        raise ValueError(
-            'The default locale "{}" is not included in the set of supported locales "{}"'.format(
-                config.DEFAULT_LOCALE, config.SUPPORTED_LOCALES
-            )
+    # These locales are available and loadable from the filesystem.
+    available = set(babel.list_translations())
+    available.add(Locale.parse(FALLBACK_LOCALE))
+
+    # These locales were configured via "securedrop-admin sdconfig", meaning
+    # they were present on the Admin Workstation at "securedrop-admin" runtime.
+    configured = parse_locale_set(config.SUPPORTED_LOCALES)
+
+    # The intersection of these sets is the set of locales usable by Babel.
+    usable = available & configured
+
+    missing = configured - usable
+    if missing:
+        babel.app.logger.error(
+            f'Configured locales {missing} are not in the set of usable locales {usable}'
         )
 
-    translations = babel.list_translations()
-    for locale in config.SUPPORTED_LOCALES:
-        if locale == "en_US":
-            continue
+    defaults = parse_locale_set([config.DEFAULT_LOCALE, FALLBACK_LOCALE])
+    if not defaults & usable:
+        raise ValueError(
+            f'None of the default locales {defaults} are in the set of usable locales {usable}'
+        )
 
-        parsed = Locale.parse(locale)
-        if parsed not in translations:
-            raise ValueError(
-                'Configured locale "{}" is not in the set of translated locales "{}"'.format(
-                    parsed, translations
-                )
-            )
+    global USABLE_LOCALES
+    USABLE_LOCALES = usable
 
 
+# TODO(#6420): avoid relying on and manipulating on this global state
 LOCALES = collections.OrderedDict()  # type: collections.OrderedDict[str, RequestLocaleInfo]
+USABLE_LOCALES = set()  # type: Set[Locale]
 
 
 def map_locale_display_names(config: SDConfig) -> None:
@@ -185,6 +199,28 @@ def configure(config: SDConfig, app: Flask) -> None:
     map_locale_display_names(config)
 
 
+def resolve_fallback_locale(config: SDConfig) -> str:
+    """
+    Return a *usable* fallback locale.  Namely:
+
+    1. Don't fall back to the configured `DEFAULT_LOCALE` if it isn't available.
+
+    2. Don't fall back to the hard-coded `FALLBACK_LOCALE` (`en_US`) if it isn't configured.
+
+    NB. If neither the default nor the fallback locale is usable, then we should have crashed
+    already in `validate_locale_configuration()`.
+    """
+
+    if Locale.parse(config.DEFAULT_LOCALE) in USABLE_LOCALES:
+        return config.DEFAULT_LOCALE
+
+    elif Locale.parse(FALLBACK_LOCALE) in USABLE_LOCALES:
+        return FALLBACK_LOCALE
+
+    else:
+        raise ValueError('No usable fallback locale')
+
+
 def get_locale(config: SDConfig) -> str:
     """
     Return the best supported locale for a request.
@@ -208,8 +244,8 @@ def get_locale(config: SDConfig) -> str:
     if not locale:
         locale = negotiate_locale(get_accepted_languages(), LOCALES.keys())
 
-    # Finally, fall back to the default locale if necessary.
-    return locale or config.DEFAULT_LOCALE
+    # Finally, if we can't negotiate a requested locale, resolve a fallback.
+    return locale or resolve_fallback_locale(config)
 
 
 def get_accepted_languages() -> List[str]:

--- a/securedrop/sdconfig.py
+++ b/securedrop/sdconfig.py
@@ -8,6 +8,9 @@ import config as _config
 from typing import Set
 
 
+FALLBACK_LOCALE = "en_US"
+
+
 class SDConfig:
     def __init__(self) -> None:
         try:
@@ -120,7 +123,7 @@ class SDConfig:
         # Config entries used by i18n.py
         # Use en_US as the default locale if the key is not defined in _config
         self.DEFAULT_LOCALE = getattr(
-            _config, "DEFAULT_LOCALE", "en_US"
+            _config, "DEFAULT_LOCALE", FALLBACK_LOCALE,
         )  # type: str
         supported_locales = set(getattr(
             _config, "SUPPORTED_LOCALES", [self.DEFAULT_LOCALE]

--- a/securedrop/tests/test_i18n.py
+++ b/securedrop/tests/test_i18n.py
@@ -32,11 +32,15 @@ from flask import render_template_string
 from flask import request
 from flask import session
 from flask_babel import gettext
-from sdconfig import SDConfig
+from i18n import parse_locale_set, resolve_fallback_locale
+from sdconfig import FALLBACK_LOCALE, SDConfig
 from sh import pybabel
 from sh import sed
 from .utils.env import TESTS_DIR
 from werkzeug.datastructures import Headers
+
+
+NEVER_LOCALE = 'eo'  # Esperanto
 
 
 def verify_i18n(app):
@@ -225,12 +229,79 @@ def test_i18n(journalist_app, config):
         verify_i18n(app)
 
 
-def test_supported_locales(config):
+def test_parse_locale_set():
+    assert parse_locale_set([FALLBACK_LOCALE]) == set([Locale.parse(FALLBACK_LOCALE)])
+
+
+def test_resolve_fallback_locale(config):
+    """
+    Only a usable default or fallback locale is returned.
+    """
+    i18n.USABLE_LOCALES = parse_locale_set([FALLBACK_LOCALE, 'es_ES'])
     fake_config = SDConfig()
 
-    # Check that an invalid locale raises an error during app
-    # configuration.
-    fake_config.SUPPORTED_LOCALES = ['en_US', 'yy_ZZ']
+    # The default locale is neither configured nor available.
+    fake_config.DEFAULT_LOCALE = NEVER_LOCALE
+    assert resolve_fallback_locale(fake_config) == FALLBACK_LOCALE
+
+    # The default locale is configured but not available.
+    fake_config.SUPPORTED_LOCALES = [FALLBACK_LOCALE, NEVER_LOCALE]
+    assert resolve_fallback_locale(fake_config) == FALLBACK_LOCALE
+
+    # The default locale is available but not configured.
+    fake_config.SUPPORTED_LOCALES = [FALLBACK_LOCALE]
+    fake_config.DEFAULT_LOCALE = NEVER_LOCALE
+    assert resolve_fallback_locale(fake_config) == FALLBACK_LOCALE
+
+    # Happy path: a non-fallback default locale is both available and configured.
+    fake_config.SUPPORTED_LOCALES = [FALLBACK_LOCALE, 'es_ES']
+    fake_config.DEFAULT_LOCALE = 'es_ES'
+    assert resolve_fallback_locale(fake_config) == 'es_ES'
+
+
+def test_no_usable_fallback_locale(journalist_app, config):
+    """
+    The apps fail if neither the default nor the fallback locale is usable.
+    """
+    fake_config = SDConfig()
+    fake_config.DEFAULT_LOCALE = NEVER_LOCALE
+    fake_config.SUPPORTED_LOCALES = [NEVER_LOCALE]
+    fake_config.TRANSLATION_DIRS = Path(config.TEMP_DIR)
+
+    i18n.USABLE_LOCALES = set()
+    with pytest.raises(ValueError, match='No usable fallback locale'):
+        resolve_fallback_locale(fake_config)
+
+    with pytest.raises(ValueError, match='in the set of usable locales'):
+        journalist_app_module.create_app(fake_config)
+
+    with pytest.raises(ValueError, match='in the set of usable locales'):
+        source_app.create_app(fake_config)
+
+
+def test_unusable_default_but_usable_fallback_locale(config, caplog):
+    """
+    The apps start even if the default locale is unusable, as along as the fallback locale is
+    usable, but log an error for OSSEC to pick up.
+    """
+    fake_config = SDConfig()
+    fake_config.DEFAULT_LOCALE = NEVER_LOCALE
+    fake_config.SUPPORTED_LOCALES = [NEVER_LOCALE, FALLBACK_LOCALE]
+    fake_config.TRANSLATION_DIRS = Path(config.TEMP_DIR)
+
+    for app in (journalist_app_module.create_app(fake_config),
+                source_app.create_app(fake_config)):
+        with app.app_context():
+            assert NEVER_LOCALE in caplog.text
+            assert 'not in the set of usable locales' in caplog.text
+
+
+def test_invalid_locales(config):
+    """
+    An invalid locale raises an error during app configuration.
+    """
+    fake_config = SDConfig()
+    fake_config.SUPPORTED_LOCALES = [FALLBACK_LOCALE, 'yy_ZZ']
     fake_config.TRANSLATION_DIRS = Path(config.TEMP_DIR)
 
     with pytest.raises(UnknownLocaleError):
@@ -239,16 +310,22 @@ def test_supported_locales(config):
     with pytest.raises(UnknownLocaleError):
         source_app.create_app(fake_config)
 
-    # Check that a valid but unsupported locale raises an error during
-    # app configuration.
-    fake_config.SUPPORTED_LOCALES = ['en_US', 'wae_CH']
+
+def test_valid_but_unusable_locales(config, caplog):
+    """
+    The apps start with one or more unusable, but still valid, locales, but log an error for
+    OSSEC to pick up.
+    """
+    fake_config = SDConfig()
+
+    fake_config.SUPPORTED_LOCALES = [FALLBACK_LOCALE, 'wae_CH']
     fake_config.TRANSLATION_DIRS = Path(config.TEMP_DIR)
 
-    with pytest.raises(ValueError, match="not in the set of translated locales"):
-        journalist_app_module.create_app(fake_config)
-
-    with pytest.raises(ValueError, match="not in the set of translated locales"):
-        source_app.create_app(fake_config)
+    for app in (journalist_app_module.create_app(fake_config),
+                source_app.create_app(fake_config)):
+        with app.app_context():
+            assert 'wae' in caplog.text
+            assert 'not in the set of usable locales' in caplog.text
 
 
 def test_language_tags():

